src/common/term_serializer.hpp

```diff
@@ -308,8 +312,9 @@ class term_serializer {
     void write_encoded_string(buffer_t &bytes, const std::string &str);
     void write_all_header(buffer_t &bytes, const term t);
 
-    term read(const buffer_t &bytes, size_t n,
-	      size_t &offset, size_t &old_hdr_size, size_t &new_hdr_size);
+    term read(const buffer_t &bytes, size_t n, size_t heap_start,
+	      std::vector<bool> &used, size_t &offset,
```

Comment:
Not at this comment, but std::vector<bool> is probably not efficiently implemented (but I'm unsure). But boost::dynamic_bitset is guaranteed to be a bit vector.